### DISCUSSION TITLE
Expansion des relations entre éléments

### DISCUSSION THREAD

**Discussion by Guerisan** [dbarzin/mercator#281] [Ideas]
Bonjour Didier, 

Encore merci pour Mercator, qui continue d'être un outil de référence  dans notre entreprise !
En allant plus loin dans la représentation du SI, nous sommes toutefois tombés sur quelques limites, principalement au niveau des vues d'infrastructures :
- Il n'est actuellement pas possible de relier plusieurs commutateurs réseaux entre eux, ainsi qu'à des routers.
- Il n'y a pas de lien possible entre commutateurs réseaux et switches physiques.
- Au niveau applicatif, une base de données ne peut être liée à sa propre VM (ce qui est le cas dans notre SI, où l'application et la base de données ne sont pas toujours sur la même machine)

Toutes ces relations ne sont effectivement pas représentées dans le guide de l'ANSSI. Mais que penses-tu de leur pertinence ?
Pour les switches, proposer par exemple une relation Many-to-Many entre eux ? Et un Many-to One avec les routers ?

**Comment by dbarzin** [comment]
Bonjour Guerisan,

Il est possible d'ajouter un objet "lien physique" entre deux objets de l'infrastructure physique.
Ce n'est pas facile à modéliser au niveau base de données car un cable n'a pas de direction et il faut éviter les doublons.
Cet objet n'aurait pas de nom mais un numéro de port source et port destination sur les deux équipements.
On peut ajouter une nouvelle vue avec des "subgraph" pour représenter les salles et les racks ainsi que de nouveaux liens et un nouveau filtre sur les liens physiques dans l'explorateur.

Pour les base de données, il faut probablement ajouter une relation "Many-to-Many" entre bases de données et serveurs logiques.

**Reply by Guerisan** [reply]
Cette idée de lien physique est intéressante. Avec un nommage des ports de connexion à la création de l'objet "câble" ?

Mais cela ne serait pas sortir de la vocation de Mercator, qui deviendrait un outil de schéma réseaux ?

Et oui pour les bases de données, la relation en many-to-many avec les serveurs logiques donnerait la souplesse nécessaire.

**Reply by dbarzin** [reply]
L'utilisation de l'objet "câble" sera facultatif, ceux qui ne souhaitent pas l'utiliser doivent bloquer son accès dans l'écran de configuration des droits d'accès. Il y une une nouvelle vue de l'infrastructure physique avec les câbles entre lé équipements.

Le lien entre bases de données et serveurs logique est prêt.

**Reply by Guerisan** [reply]
Voilà qui me semble un bon compromis !

Parfait pour le lien des bases de données. Merci beaucoup :)

**Reply by dbarzin** [reply]
Cela avance bien, il y aura une première version du diagrame réseau avec les liens physique en fin de semaine.

**Reply by dbarzin** [reply]
Un lien physique entre équipement réseau a été ajouté et la vue de l’infrastructure réseau est en place et fonctionne bien. Il faudrait maintenant mettre à jour "l'explorateur" d'objets...

**Reply by Guerisan** [reply]
Pardon pour le délai de réponse !
Et merci pour l'implémentation :)
L'explorateur d'objet permet en principe déjà de visualiser tous les items de mercator.
Il y faut y rajouter le lien ^physique ?

**Reply by dbarzin** [reply]
Il y a un nouvel objet PhysicalLink et une nouvelle vue de l'infrastructure réseau qui permettent de faire le lien entre objet physique et virtuel du point de vue réseau. Il reste à mettre l'explorateur à jour.

**Reply by dbarzin** [reply]
C'est en place !

**Reply by Guerisan** [reply]
Top !
La vue réseau est bienvenue, et fonctionne bien !
Mais je n'ai pas encore pu tester les liens physiques, les vues semblent absentes (index et create)
![image](https://user-images.githubusercontent.com/43884015/217520237-414e67a1-3558-401a-8c25-c3fa52b5ca7c.png)
Tu aurais une idée ?

**Comment by dbarzin** [comment]
Problème de cache. Essaye :

    php artisan optimize:clear

**Reply by Guerisan** [reply]
Même souci, malgré un rebuild du docker :/

**Reply by dbarzin** [reply]
Comment fais-tu pour déployer Mercator avec Docker ?
C'est une question qui a été posée plusieurs fois, pourrais-tu faire une petite documentation ?
Merci,
Didier

**Reply by Guerisan** [reply]
Bien sûr, je proposerai ça rapidement

**Reply by Guerisan** [reply]
Mais je me suis principalement basé sur ta documentation

**Reply by Guerisan** [reply]
[Docker mercator.md](https://github.com/dbarzin/mercator/files/10698386/Docker.mercator.md)

Les seules additions concernent les soucis de synchronisation du fichier ssmtp.conf (volume par défaut dans le docker-compose.yml), et l'encodage des retours chariots du fichier endpoint.sh, qui a posé souci à des collègues sous windows

**Comment by dbarzin** [comment]
OK, les pages étaient manquantes.
Tu peux faire un git pull.

**Reply by Guerisan** [reply]
Tout est OK !
Merci beaucoup !
